sonar-core/src/main/java/org/sonar/core/issue/tracking/LineHashSequence.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.core.issue.tracking;
+
+import com.google.common.base.Strings;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.commons.lang.StringUtils;
+
+/**
+ * Sequence of hash of lines for a given file
+ */
+public class LineHashSequence {
+
+  private static final int[] EMPTY_INTS = new int[0];
+
+  /**
+   * Hashes of lines. Line 1 is at index 0. No null elements.
+   */
+  private final List<String> hashes;
+  private final Map<String, int[]> linesByHash;
+
+  public LineHashSequence(List<String> hashes) {
```

Comment:
you could use a ImmutableMapBuilder (this way you don't need to care about the size of the collection argument) and take an Iterable instead of a List to support building this object from a stream of lines
in addition, this argument should be called lines if I'm not mistaken